services/core-web/src/components/common/CoreTableCommonColumns.tsx

```diff
@@ -74,103 +72,57 @@ export const renderHighlightedTextColumn = (
   };
 };
 
-const documentWithTag = (record: any, elem: ReactNode, title: string) => {
-  return (
-    <div
-      className="inline-flex flex-between file-name-container"
-      style={!record.number_of_versions ? { marginLeft: "0" } : { marginLeft: "14px" }}
-      title={title}
-    >
-      {elem}
+export interface ITableAction {
+  key: string;
+  label: string;
+  clickFunction: (event, record) => any;
+  icon?: ReactNode;
+}
 
-      <span className="file-history-container">
-        {record?.number_of_versions > 0 ? (
-          <span>
-            <Tooltip
-              title={`This file has ${record.number_of_versions} previous versions`}
-              placement="top"
-              mouseEnterDelay={1}
-            >
-              <Tag icon={<ClockCircleOutlined />} color="#5E46A1" className="file-version-amount">
-                {record.number_of_versions}
-              </Tag>
-            </Tooltip>
-          </span>
-        ) : null}
-        {record.showArchiveIndicator && record?.is_archived ? <Tag>{"Archived"}</Tag> : null}
-      </span>
-    </div>
-  );
-};
-
-export const renderDocumentLinkColumn = (
-  dataIndex: string,
-  title = "File Name",
-  sortable = true,
-  docManGuidIndex = "document_manager_guid",
-  showArchiveIndicator = true
-): ColumnType<any> => {
-  return {
-    title,
-    dataIndex,
-    key: dataIndex,
-    render: (text = "", record: any) => {
-      const recordWithArchiveIndicator = { ...record, showArchiveIndicator };
-      const link = (
-        <div
-          key={record.key ?? record[docManGuidIndex]}
-          className={record.number_of_versions !== undefined ? "file-name-text" : ""}
-          style={record?.number_of_versions === 0 ? { marginLeft: "38px" } : {}}
-        >
-          <DocumentLink
-            documentManagerGuid={record[docManGuidIndex]}
-            documentName={text}
-            truncateDocumentName={false}
-          />
-        </div>
-      );
-      return documentWithTag(recordWithArchiveIndicator, link, title);
-    },
-    ...(sortable ? { sorter: nullableStringSorter(dataIndex) } : null),
-  };
-};
-
-export const renderTaggedColumn = (
-  dataIndex: string,
-  title: string,
-  sortable = false,
-  placeHolder = ""
+export const renderActionsColumn = (
```

Comment:
I moved all the document-specific columns to DocumentColumns.

Put the interface here because the only reason to use it at all is this function (so importing both). Would be willing to move it though)